src/core/dom/layout/intersection-no-root.js

```diff
@@ -40,15 +41,15 @@ function getInOb(win) {
           }
           seen.add(target);
 
-          observer.unobserve(target);
-          intersectionDeferreds.get(target).resolve(entries[i]);
-          intersectionDeferreds.delete(target);
+          devAssert(observer).unobserve(target);
```

Comment:
`devAsserts` should be DCE'd away without exception since `mode.isMinified()` is true in prod. In every assertion function in `#core/assert/dev`, we have a call to `devAssertDceCheck` which contains a sentinel value. One test we run post-build is to ensure that sentinel value isn't present in v0.js, so we can be confident it's getting DCE'd away.

Reply:
Most of the body gets stripped, but the closure (and unused params!) stick around.

Reply:
Do calls like this stick around? If so that's a huge regression

Reply:
![image](https://user-images.githubusercontent.com/6694512/145445319-40e79148-d331-47f3-8e21-c856d8ea7a65.png)

This message doesn't appear anywhere in v0.js. The code compiles to:
```javascript
g.registerDefaultAction=function(a,b,c){var d=void 0===b?"activate":b;this.registerAction(d,a,void 0===c?2:c);this.defaultActionAlias_=d};
```

I'm not certain what you're referencing (perhaps userAsserts?) but `devAssert(foo)` gets DCE'd to `foo` correctly AFAICT

Reply:
The case like this where the value isn't used is DCEd fine, but the case where the resulting value is used I don't think does. I'll find an example